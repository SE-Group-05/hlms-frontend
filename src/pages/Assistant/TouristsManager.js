import React, { useEffect, useState } from 'react';
import { AddIcon } from "@chakra-ui/icons";
import { Box, Center, Heading, Button, useDisclosure, Modal, ModalOverlay, ModalContent } from "@chakra-ui/react";
import AddTouristForm from "../../components/Forms/AddTouristsForm";
import TouristCard from "../../components/Assistant/TouristsCard";
import SearchBar from "../../components/Assistant/SearchBar";
import { touristService } from "../../services/touristService";

function TouristsManager() {
    const { isOpen, onOpen, onClose } = useDisclosure();
    const [data, setData] = useState([]);
    const [hasError, setErrors] = useState(false);

    const fetchData = async () => {
        touristService.getTourists()
            .then((result) => {
                if (result.success) {
                    setData(result.tourists);
                } else {
                    setErrors(result.status);
                }
            });
    }
    useEffect(() => {
        fetchData();
    }, []);

    return (
        <>
            <Button position="fixed" bottom="50px" right="50px" zIndex='1000' leftIcon={<AddIcon />} colorScheme="teal" onClick={onOpen}>
                Add Tourists
        </Button>
            <Modal isOpen={isOpen} onClose={onClose}>
                <ModalOverlay />
                <ModalContent top='100px'>
                    <AddTouristForm />
                </ModalContent>
            </Modal>
            <Center w="100%" h="auto" p={4} pt='180px' pb='100px'>
                <Box w="90%" h="auto" borderWidth="1px" borderRadius="lg" p='30px'>
                    <Heading ml='20px' mb='20px'>Tourist Manager</Heading>
                    <Center >
                        <SearchBar text='Search Tourists...' />
                    </Center>
                    {data.map((user) => (
                        <TouristCard
                            key={user.id}
                            id={user.id}
                            imageUrl="./user.png"
                            first_name={user.firstname}
                            last_name={user.lastname}
                            email={user.email}
                            phone={user.phone}

                        />
                    ))}

                </Box>
            </Center>
        </>
    )
}

export default TouristsManager;