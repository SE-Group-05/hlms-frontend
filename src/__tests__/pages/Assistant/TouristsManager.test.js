// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import TouristsManager from "../../../pages/Assistant/TouristsManager";
import { MemoryRouter } from "react-router-dom";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter><TouristsManager /></MemoryRouter>,
    div
  );
});

it("search button works correctly", async () => {
    const { queryByTestId, getByText } = render(
        <MemoryRouter><TouristsManager /></MemoryRouter>
    );
    fireEvent.change(queryByTestId("search"),{ target: { value: 'mirissa' } })
    fireEvent.click(getByText('Add Tourists'))
  });
  


it("matches snapshot", () => {
  const tree = renderer.create(
    <MemoryRouter><TouristsManager /></MemoryRouter>
  );

  expect(tree).toMatchSnapshot();
});
