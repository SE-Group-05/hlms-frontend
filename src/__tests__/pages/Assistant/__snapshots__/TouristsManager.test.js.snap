// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <button
    className="chakra-button css-owb5xr"
    onClick={[Function]}
    type="button"
  >
    <span
      className="chakra-button__icon css-ktvjz1"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-onkibi"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
          fill="currentColor"
        />
      </svg>
    </span>
    Add Tourists
  </button>,
  <div
    className="css-nr45wa"
  >
    <div
      className="css-102niab"
    >
      <h2
        className="chakra-heading css-1h40200"
      >
        Tourist Manager
      </h2>
      <div
        className="css-gmuwbf"
      >
        <div
          className="css-1ws0jr4"
        >
          <div
            className="chakra-input__group css-4302v8"
          >
            <div
              className="chakra-input__right-element css-a6fysx"
            >
              <button
                aria-label="Search database"
                className="chakra-button css-1kgd4fd"
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="chakra-icon css-onkibi"
                  focusable={false}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                    fill="currentColor"
                  />
                </svg>
              </button>
            </div>
            <input
              className="chakra-input css-0"
              data-testid="search"
              name="search"
              onBlur={[Function]}
              onFocus={[Function]}
              placeholder="Search Tourists..."
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
